site/content/docs/howto/configure-supervisor-with-auth0.md

```diff
@@ -0,0 +1,147 @@
+## Pass user groups into the token for K8s RBAC
+
+There are a handful of ways to include user groups into the Auth0 ID token so that Kubernetes RBAC can be used to assign permissions on the cluster.
+
+**NOTE**: Pinniped does not have a specific recommendation for how user groups are defined in Auth0.
+The examples below are provided as examples to better understand how Auth0 and Pinniped integrate, not how to configure Auth0.
+
+One way to add user groups into the ID token is to add a [custom claim](https://auth0.com/blog/adding-custom-claims-to-id-token-with-auth0-actions/).
+Auth0 recommends using a [namespaced format](https://auth0.com/docs/secure/tokens/json-web-tokens/create-custom-claims) for your claim names.
+Pinniped requires that the group claim is an array of strings.
+
+In this example, replace `"https://example.com/pinniped/groups"` with the namespaced claim name of your choice.
+The array of groups should be dynamically provisioned from the Auth0 user store.
+
+```typescript
+exports.onExecutePostLogin = async (event, api) => {
+  if (event.authorization) {
+    api.idToken.setCustomClaim("https://example.com/pinniped/groups", ["auth0-read-only", "other-grouo", "something-else"]);
+  }
+};
+```
+
+After login, you can grant one of those groups access to your cluster, for example with a `ClusterRoleBinding`:
+
+```shell
+kubectl create clusterrolebinding read-only \
+    --clusterrole=view \
+    --group=auth0-read-only
+```
+
+## Configure the Supervisor
+
+Create an [OIDCIdentityProvider](https://github.com/vmware-tanzu/pinniped/blob/main/generated/{{< latestcodegenversion >}}/README.adoc#oidcidentityprovider) in the same namespace as the Supervisor.
+
+For example, this OIDCIdentityProvider uses Auth0's `email` claim as the Kubernetes username:
+
+```yaml
+apiVersion: idp.supervisor.pinniped.dev/v1alpha1
+kind: OIDCIdentityProvider
+metadata:
+  namespace: pinniped-supervisor
+  name: auth0
+spec:
+
+  # Change this to be the actual issuer provided by your Auth0 account.
+  issuer: https://<your-tenant-id>.<your-region>.auth0.com/
+
+  authorizationConfig:
+
+    # Request any scopes other than "openid" for claims besides
+    # the default claims in your token. The "openid" scope is always
+    # included.
+    additionalScopes: 
+    - offline_access
+    - email
+
+    # If you would also like to allow your end users to authenticate using
+    # a password grant, then change this to true. Password grants only work
+    # with applications created in Auth0 with the "Password" grant type enabled.
+    allowPasswordGrant: false
+
+  # Specify how Auth claims are mapped to Kubernetes identities.
+  claims:
+
+    # Specify the name of the claim in your Auth0 token that will be mapped
+    # to the "username" claim in downstream tokens minted by the Supervisor.
+    username: email
+
+    # Specify the name of the claim in Auth0 that represents the groups
+    # that the user belongs to.
+    groups: https://<your-domain>/groups
```

Comment:
Should this be the same as the example provided above? `"https://example.com/pinniped/groups"`

Reply:
done, see next push